keras/engine/topology.py

```diff
@@ -2815,7 +2818,7 @@ def preprocess_weights_for_loading(layer, weights,
             weights[0] = weights[0][:, 0, :, :]
 
         if layer.__class__.__name__ == 'Conv2D':
-            if layer.data_format == 'channels_first':
+            if layer.data_format == 'channels_first' and original_backend in {'th', 'theano'}:
```

Comment:
@fchollet It doesn't seem to know whether the old model had dim_ordering was 'tf' or 'th', and transposing is not right if it was 'tf':
https://github.com/fchollet/keras/blob/1.2.2/keras/layers/convolutional.py#L422
Transposing in this case messes things up and is the main source of failure.